server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -282,13 +283,16 @@ public void write(OutputStream outputStream) throws IOException, WebApplicationE
          .addData("peer", req.getRemoteAddr())
          .emit();
 
-      return Response.serverError().type(contentType).entity(
-          jsonWriter.writeValueAsBytes(
-              ImmutableMap.of(
-                  "error", e.getMessage() == null ? "null exception" : e.getMessage()
-              )
-          )
-      ).build();
+      return Response.serverError().type(contentType).entity(cleanError(e)).build();
     }
   }
+
+  private byte[] cleanError(Exception e) throws JsonProcessingException
+  {
+    return jsonMapper.writeValueAsBytes(
```

Comment:
This actually changes the behavior, using jsonMapper instead of jsonWriter, which is incorrect when returning smile data. Any reason this was put in a separate method?